lucene/core/src/java/org/apache/lucene/index/DocumentsWriter.java

```diff
@@ -322,35 +316,27 @@ synchronized Closeable lockAndAbortAll() throws IOException {
   }
   
   /** Returns how many documents were aborted. */
-  private int abortThreadState(final ThreadState perThread) throws IOException {
+  private int abortDocumentsWriterPerThread(final DocumentsWriterPerThread perThread) throws IOException {
     assert perThread.isHeldByCurrentThread();
-    if (perThread.isInitialized()) { 
-      try {
-        int abortedDocCount = perThread.dwpt.getNumDocsInRAM();
-        subtractFlushedNumDocs(abortedDocCount);
-        perThread.dwpt.abort();
-        return abortedDocCount;
-      } finally {
-        flushControl.doOnAbort(perThread);
-      }
-    } else {
+    try {
+      int abortedDocCount = perThread.getNumDocsInRAM();
+      subtractFlushedNumDocs(abortedDocCount);
+      perThread.abort();
+      return abortedDocCount;
+    } finally {
       flushControl.doOnAbort(perThread);
-      // This DWPT was never initialized so it has no indexed documents:
-      return 0;
     }
   }
 
   /** returns the maximum sequence number for all previously completed operations */
   public long getMaxCompletedSequenceNumber() {
-    long value = lastSeqNo;
-    int limit = perThreadPool.getMaxThreadStates();
-    for(int i = 0; i < limit; i++) {
-      ThreadState perThread = perThreadPool.getThreadState(i);
-      value = Math.max(value, perThread.lastSeqNo);
-    }
-    return value;
+    // NOCOMMIT: speak to mikemccandless about this change https://github.com/apache/lucene-solr/commit/5a03216/
+    // Returning the last seqNum is as good as the way we had before IMO. I tried to figure out why this is better but
+    // failed.
```

Comment:
I left a (non-review) comment on the single commit about this.

I think it's a good tradeoff?  Instead of making each indexing op compute `max` to save `lastSeqNo`, we are making caller of this API do the legwork.  I think it's a fair tradeoff since likely this API is very rarely called, but indexing ops are very often called.

Reply:
hmm, I am not sure I understand fully. You mean the way it used to be was a good tradeoff? Currently there is no extra work done after my change, I am struggeling to understand why you made the chance in the first place. To me it's equivalent and no tests are failing.

Reply:
I am saying your change is the better tradeoff than the way it was before.

Reply:
cool thanks for clarifying